pymodbus/framer/socket_framer.py

```diff
@@ -36,13 +37,32 @@ class ModbusSocketFramer(ModbusFramer):
 
     method = "socket"
 
-    def __init__(self, decoder, client=None):
+    def __init__(self, decoder, **kwargs):
         """Initialize a new instance of the framer.
 
         :param decoder: The decoder factory implementation to use
         """
-        super().__init__(decoder, client)
+        super().__init__(decoder, **kwargs)
         self._hsize = 0x07
+        self.transport = kwargs.get('transport')
+
+    def _validate_slave_id(self, slaves: list, single: bool) -> bool:
+        """Validate if the received data is valid for the client.
+
```

Comment:
having _ means it´s an internal function, but I cannot see that it is called anywhere.

Reply:
It is overwriting the method from [ModbusFramer](https://github.com/pymodbus-dev/pymodbus/blob/131d30391efd9c64f1fec857ae0372746374e6d2/pymodbus/framer/base.py#L48C1-L61C45).

Reply:
But that is wrong, what happens when I use the rtu framer on tcp.

Or the tls framer for that matter.